kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -795,6 +797,44 @@ def foundry_section_edge(
     apr_proof.write_proof()
 
 
+def foundry_get_model(
```

Comment:
This function should produce a data structure (maybe `FoundryModel | None`). Function `print_model` can be a method on this class (without the `"Failed to ..."` branch). Function `print_failure_info` then can be redefined as

```python3
def print_failure_info(kprint: KPrint, proof: Proof, *, model: FoundryModel | None = None) -> list[str]:
    ...
```

Reply:
Thanks, I agree! Will add the `FoundryModel` class and modify the printing functions accordingly (there's some more refactoring to be done, as suggested by @nwatson22), but I need a bit more time to work on it. Would you recommend to merge the current PR and refactor it later, or should I include this change in this PR? Thank you!

Reply:
It's fine either way. If you decide to merge now, please open an issue to document the requested changes.

Reply:
Issue created: https://github.com/runtimeverification/evm-semantics/issues/1955